core/state/statedb.go

```diff
@@ -933,6 +934,19 @@ func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, error) {
 				storageTrieNodesUpdated += updates
 				storageTrieNodesDeleted += deleted
 			}
+		} else {
+			// Account is deleted, nuke out the storage data as well.
+			set, err := obj.DeleteTrie(s.db)
```

Comment:
If we merge this PR, the production code will start calling `obj.DeleteTrie`. It will soup up all the destructed storage slots into `set` , which get merged into `nodes`, 

I think it will eventually get ignored in `trie.Database.Update`. Is that correct?

Reply:
Yes.

Reply:
Good idea. I'll start one

Reply:
This is now running on `03` against master on `04`

Reply:
```
INFO [10-19|07:25:02.027] Imported new chain segment               blocks=46   txs=6636    mgas=572.624  elapsed=8.215s       mgasps=69.700  number=10,582,621 hash=2c6a5f..1cd05a age=2y2mo3w   dirty=1022.53MiB
WARN [10-19|07:25:05.512] Full stats report failed                 err="use of closed network connection"
INFO [10-19|07:25:10.056] Imported new chain segment               blocks=54   txs=8816    mgas=633.603  elapsed=8.028s       mgasps=78.920  number=10,582,675 hash=222132..b6c4c3 age=2y2mo3w   dirty=1023.59MiB
INFO [10-19|07:25:11.136] Forkchoice requested sync to new head    number=15,780,646 hash=838d5d..d2a0dc
INFO [10-19|07:25:18.204] Imported new chain segment               blocks=38   txs=6031    mgas=474.161  elapsed=8.148s       mgasps=58.191  number=10,582,713 hash=5e092c..6122bc age=2y2mo3w   dirty=1022.13MiB
INFO [10-19|07:25:26.255] Imported new chain segment               blocks=38   txs=6154    mgas=449.242  elapsed=8.051s       mgasps=55.799  number=10,582,751 hash=81bdcc..5b863f age=2y2mo3w   dirty=1022.24MiB
INFO [10-19|07:25:34.351] Imported new chain segment               blocks=49   txs=7418    mgas=560.476  elapsed=8.096s       mgasps=69.227  number=10,582,800 hash=6d1fac..ab53b9 age=2y2mo3w   dirty=1022.47MiB
INFO [10-19|07:25:42.389] Imported new chain segment               blocks=37   txs=5029    mgas=449.420  elapsed=8.037s       mgasps=55.912  number=10,582,837 hash=e49835..28dbce age=2y2mo3w   dirty=1020.94MiB
INFO [10-19|07:25:50.502] Imported new chain segment               blocks=43   txs=5340    mgas=520.497  elapsed=8.112s       mgasps=64.162  number=10,582,880 hash=da883b..febd87 age=2y2mo3w   dirty=1019.56MiB
INFO [10-19|07:25:58.552] Imported new chain segment               blocks=40   txs=5124    mgas=483.557  elapsed=8.050s       mgasps=60.066  number=10,582,920 hash=868f6b..d04aea age=2y2mo3w   dirty=1023.95MiB
INFO [10-19|07:26:06.656] Imported new chain segment               blocks=39   txs=5016    mgas=484.501  elapsed=8.104s       mgasps=59.782  number=10,582,959 hash=04c5ce..ecf819 age=2y2mo3w   dirty=1023.16MiB
INFO [10-19|07:26:14.831] Imported new chain segment               blocks=40   txs=4348    mgas=472.555  elapsed=8.174s       mgasps=57.806  number=10,582,999 hash=2e0993..e5e54e age=2y2mo3w   dirty=1022.21MiB
INFO [10-19|07:26:22.876] Imported new chain segment               blocks=55   txs=5907    mgas=657.391  elapsed=8.044s       mgasps=81.715  number=10,583,054 hash=850a06..fe8467 age=2y2mo3w   dirty=1020.27MiB
INFO [10-19|07:26:31.040] Imported new chain segment               blocks=37   txs=4889    mgas=460.583  elapsed=8.164s       mgasps=56.414  number=10,583,091 hash=14c569..1203d0 age=2y2mo3w   dirty=1022.72MiB
INFO [10-19|07:26:39.110] Imported new chain segment               blocks=37   txs=4654    mgas=459.029  elapsed=8.069s       mgasps=56.887  number=10,583,128 hash=df3e70..8ca1d7 age=2y2mo3w   dirty=1023.03MiB
INFO [10-19|07:26:47.147] Imported new chain segment               blocks=39   txs=4002    mgas=472.347  elapsed=8.037s       mgasps=58.766  number=10,583,167 hash=97cd6f..31b5e0 age=2y2mo3w   dirty=1023.31MiB
ubuntu@bench03:~$
```

Should be ok after running 6 days

Reply:
actually.. was just looking into this, and I don't think I ran the correct code

Reply:
restarted. sorry :(

Reply:
no worry, let's wait a few more days then